test/mocks/http/mocks.h

```diff
@@ -258,6 +259,7 @@ class MockStreamDecoderFilterCallbacks : public StreamDecoderFilterCallbacks,
   MOCK_METHOD1(encodeTrailers_, void(HeaderMap& trailers));
 
   Buffer::InstancePtr buffer_;
+  std::list<DownstreamWatermarkCallbacks*> callbacks_{};
```

Comment:
Since we don't currently support list/vector in real code should we do it here? Perhaps switch with pointer and appropriate asserts?